org.palladiosimulator.analyzer.slingshot.behavior.spd/src/org/palladiosimulator/analyzer/slingshot/behavior/spd/interpreter/utils/TargetGroupUtils.java

```diff
@@ -1,15 +1,41 @@
 package org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.utils;
 
+import org.palladiosimulator.analyzer.slingshot.core.Slingshot;
+import org.palladiosimulator.pcm.allocation.Allocation;
 import org.palladiosimulator.pcm.resourceenvironment.ResourceContainer;
 import org.palladiosimulator.spd.targets.ElasticInfrastructure;
+import org.palladiosimulator.spd.targets.ServiceGroup;
+import org.palladiosimulator.spd.targets.TargetGroup;
 
 public class TargetGroupUtils {
 	
+	private static final Allocation allocation = Slingshot.getInstance().getInstance(Allocation.class);
+	
 	public static boolean isContainerInElasticInfrastructure(final ResourceContainer container, final ElasticInfrastructure targetGroup) {
 		return targetGroup.getPCM_ResourceEnvironment()
 						  .getResourceContainer_ResourceEnvironment()
 						  .stream()
 						  .anyMatch(rc -> rc.getId().equals(container.getId()));
 	}
 	
+	public static boolean isContainerInServiceGroup(final ResourceContainer container, final ServiceGroup serviceGroup) {
+		return allocation.getAllocationContexts_Allocation()
+						 .stream()
+						 .filter(ac -> ac.getAssemblyContext_AllocationContext().getId().equals(serviceGroup.getUnitAssembly().getId()))
```

Comment:
@kjuli adapts this to retrieve allocations of all elements in the service group.